SU-SOE/stanford_soe_helper
# SOE-2667: Add department views

# READY FOR REVIEW


# Summary
For the department pages on the SoE site, they want to display magazine articles that are related to the department. This PR adds a view with 3 displays. Each display relies on the department taxonomy through the contextual filter and displays only articles for that department. 

If there's no *featured* for the department, *most recent for the department* will display. If there's  no article tagged for the department, it will just display the *most recent article* on the site.

Because this relies on the department taxonomy, this has been moved from Stanford Magazine to Stanford SoE Helper Magazine.

# Needed By (Date)
- Soon

# Criticality
- Client is eagerly awaiting this.

# Steps to Test

- Check if the _Stanford Magazine Article Views feature_ is overridden. If so, stop and let's talk.
- Fetch and checkout this branch ( 7.x-1.x-dept-views ) on to a SoE site
- Verify that the feature: _Stanford Magazine Article Views_ is in default state. If necessary you'll need to revert the feature. 
- Place the new blocks on a department page. So for example, navigate to: department/aeronautics-and-astronautics and place the blocks using context or blocks module. Doesn't matter for this testing.

The blocks to place are:
```

View: Stanford Magazine Article: Department: Block Dept Featured 
View: Stanford Magazine Article: Department: Block Dept Most Recent
View: Stanford Magazine Article: Department: Block Dept Random

```

- Go to the taxonomy page for the department, i.e. department/aeronautics-and-astronautics
- Make sure there's a featured block for the department
- Verify you know which block is most recently published for the department
- Verify that each block displays the article correctly, i.e. featured, recent, or random

# Affects 
- SoE (School of Engineering website)

# Associated Issues and/or People
## Related JIRA ticket(s)
Development:  https://stanfordits.atlassian.net/browse/SOE-2667
PR:  https://stanfordits.atlassian.net/browse/SOE-2684
## Related PRs
This obviates https://github.com/SU-SWS/stanford_magazine/pull/30
## More Information

## Folks to notify
@minorwm 
@andydunmire 


# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
